packages/react-ui/src/features/pieces/lib/piece-selector-utils.ts

```diff
@@ -0,0 +1,275 @@
+                  [
+                    {
+                      operator: BranchOperator.TEXT_EXACTLY_MATCHES,
+                      firstValue: '',
+                      secondValue: '',
+                      caseSensitive: false,
+                    },
+                  ],
+                ],
+                branchType: BranchExecutionType.CONDITION,
+                branchName: 'Branch 1',
+              },
+              {
+                branchType: BranchExecutionType.FALLBACK,
+                branchName: 'Otherwise',
+              },
+            ],
+            inputUiInfo: {
+              customizedInputs: {},
+            },
+          },
+          children: [null, null],
+        },
+        common,
+      );
+    case ActionType.PIECE: {
+      if (!isPieceActionOrTrigger(pieceSelectorItem)) {
+        throw new Error(
+          `Invalid piece selector item ${JSON.stringify(pieceSelectorItem)}`,
+        );
+      }
+      return deepMergeAndCast<PieceAction>(
+        {
+          type: ActionType.PIECE,
+          settings: settings ?? {
+            pieceName: pieceSelectorItem.pieceMetadata.pieceName,
+            pieceType: pieceSelectorItem.pieceMetadata.pieceType,
+            packageType: pieceSelectorItem.pieceMetadata.packageType,
+            actionName: pieceSelectorItem.actionOrTrigger.name,
+            pieceVersion: pieceSelectorItem.pieceMetadata.pieceVersion,
+            input,
+            errorHandlingOptions,
+          },
+        },
+        common,
+      );
+    }
+    case TriggerType.PIECE: {
+      if (!isPieceActionOrTrigger(pieceSelectorItem)) {
+        throw new Error(
+          `Invalid piece selector item ${JSON.stringify(pieceSelectorItem)}`,
+        );
+      }
+      return deepMergeAndCast<PieceTrigger>(
+        {
+          type: TriggerType.PIECE,
+          settings: settings ?? {
+            pieceName: pieceSelectorItem.pieceMetadata.pieceName,
+            pieceType: pieceSelectorItem.pieceMetadata.pieceType,
+            packageType: pieceSelectorItem.pieceMetadata.packageType,
+            triggerName: pieceSelectorItem.actionOrTrigger.name,
+            pieceVersion: pieceSelectorItem.pieceMetadata.pieceVersion,
+            input,
+          },
+        },
+        common,
+      );
+    }
+    default:
+      throw new Error('Unsupported type: ' + pieceSelectorItem.type);
+  }
+};
+
+const checkPieceInputValidity = (
+  input: Record<string, unknown>,
+  props: PiecePropertyMap,
+) => {
+  return Object.entries(props).reduce((acc, [key, property]) => {
+    if (
+      property.required &&
+      property.type !== PropertyType.DYNAMIC &&
+      isNil(input[key])
+    ) {
+      return false;
+    }
+    return acc;
+  }, true);
+};
+
+const maxListHeight = 300;
+const minListHeight = 100;
+const searchInputDivHeight = 48;
+
+// Adjusts piece list height to prevent overflow on short screens
+const useAdjustPieceListHeightToAvailableSpace = () => {
```

Comment:
Do we really need this? can it be done with css like max-height and leave it for radix to figure it out?